src/ts/utils/dom.ts

```diff
@@ -5,7 +5,23 @@ export function getActiveEditElement(): ValueElement {
 
     let element = document.activeElement;
     // on some pages like chrome://history/, input is in shadowRoot of several other recursive shadowRoots.
-    while (element?.shadowRoot) {
+    while (element && element.shadowRoot) {
+        if (element.shadowRoot.activeElement) {
+            element = element.shadowRoot.activeElement;
+        } else {
+            const subElement = element.shadowRoot.querySelector('input, textarea, select');
+            if (subElement) {
+                element = subElement;
+            }
+            break;
+        }
+    }
+    return element as ValueElement;
+}
+
+export function getTopLevelBlockList(): ValueElement {
+    let element = document.querySelector('.roam-article div .flex-v-box');
+    while (element && element.shadowRoot) {
```

Comment:
I don't think you need this stuff for the block list

Reply:
also - this is not a textarea/input/etc so should not be `ValueElement` just HTMLElement?

the list name seems to imply that you'd return list of them, how about "container"?

Reply:
yep - copypasta here :P

Reply:
though I'm not sure I understand the shadowRoot thing so that's why I left it there

Reply:
I think that one is specifically around getting real edit element from `activeElement` I'm not entirely sure about details either - stole it from Surfingkeys, need to follow up on this